SpatialGDK/Source/SpatialGDKTests/Examples/SpatialGDKExampleTest.cpp

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "TestDefinitions.h"
+
+#include "Misc/ScopeTryLock.h"
+#include "HAL/PlatformFilemanager.h"
+
+#include <memory>
+
+#define EXAMPLE_SIMPLE_TEST(TestName) \
+	GDK_TEST(SpatialGDKExamples, SimpleExamples, TestName)
+
+#define EXAMPLE_COMPLEX_TEST(TestName) \
+	GDK_COMPLEX_TEST(SpatialGDKExamples, ComplexExamples, TestName)
+
+DECLARE_LOG_CATEGORY_EXTERN(LogSpatialGDKExamples, Log, All);
+DEFINE_LOG_CATEGORY(LogSpatialGDKExamples);
+
+// 1. Latent command example
+namespace
+{
+const double MAX_WAIT_TIME_FOR_BACKGROUND_COMPUTATION = 2.0;
+const double MIN_WAIT_TIME_FOR_BACKGROUND_COMPUTATION = 1.0;
+const double COMPUTATION_DURATION = 1.0;
+
+struct
+{
+	FCriticalSection Mutex;
+	int Value;
+} ComputationResult;
+} // anonymous namespace
+
+DEFINE_LATENT_AUTOMATION_COMMAND(FStartBackgroundThreadComputation);
+bool FStartBackgroundThreadComputation::Update()
+{
+	AsyncTask(ENamedThreads::AnyBackgroundThreadNormalTask, []
+	{
+		FScopeLock BackgroundComputationLock(&ComputationResult.Mutex);
+		FPlatformProcess::Sleep(COMPUTATION_DURATION);
+		ComputationResult.Value = 42;
+	});
+
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_ONE_PARAMETER(FWaitForComputationAndCheckResult, FAutomationTestBase*, Test);
+bool FWaitForComputationAndCheckResult::Update()
+{
+	const double TimePassed = FPlatformTime::Seconds() - StartTime;
+
+	if (TimePassed >= MIN_WAIT_TIME_FOR_BACKGROUND_COMPUTATION)
+	{
+		FScopeTryLock BackgroundComputationLock(&ComputationResult.Mutex);
+
+		if (BackgroundComputationLock.IsLocked())
+		{
+			Test->TestTrue("Computation result is equal to expected value", ComputationResult.Value == 42);
+			return true;
+		}
+
+		if (TimePassed >= MAX_WAIT_TIME_FOR_BACKGROUND_COMPUTATION)
+		{
+			Test->TestTrue("Computation finished in time", false);
```

Comment:
```suggestion
			Test->TestTrue("Computation didn't finish in time", false);
```

Reply:
Unless I've misunderstood

Reply:
The idea was that `Computation finished in time` == false, hence TestTrue fails. Can change as you suggest for readability.

Reply:
Gotcha